src/airport.js

```diff
@@ -1 +1,78 @@
+class Airport {
+    listOfPlanes;
+    airportCapacity;
+    currentWeather;
+    notPlaneError = new Error("Object is not a plane");
+    atCapacityError = new Error("The airport is at capacity");
+    notAtAirportError = new Error("That plane is not at the airport");
+    capacityFormatError = new Error("Invalid capacity");
+    alreadyLandedPlaneError = new Error("Cannot land an already landed plane");
+    tooStormyError = new Error("The weather is too bad to fly");
+    constructor(initialListOfPlanes = [], initialCapacity = 3, initialWeather = "Sunny") {
+        this.listOfPlanes = initialListOfPlanes;
+        this.airportCapacity = initialCapacity;
+        this.currentWeather = initialWeather;
+    };
+    isPlaneError(plane) {
+        if (!plane?.planeId) {
+            throw this.notPlaneError;
+        }
+    };
+    isAtCapacityError() {
+        if (this.isAirportFull()) {
+            throw this.atCapacityError;
+        }
+    };
+    isNotInAirportError(plane) {
+        if (!this.isPlaneInAirport(plane)) {
+            throw this.notAtAirportError;
+        }
+    };
+    isCapacityFormatError(newCapacity) {
+        if (!(typeof newCapacity === "number" && Number.isInteger(newCapacity))) {
+            throw this.capacityFormatError;
```

Comment:
Should choose better names for the error check methods. e.g. starting with "check"